### PR TITLE
docs(hermetic-build): indicate usage of Docker Buildkit in development guide

### PR DESCRIPTION
**Pull request by diegomarquezp**

### CODE REVIEW

File: hermetic_build/DEVELOPMENT.md [hunk 1]
```diff
@@ -160,13 +160,15 @@ python hermetic_build/library_generation/cli/entry_point.py generate \
...
+   DOCKER_BUILDKIT=1 docker build \
      -f .cloudbuild/library_generation/library_generation.Dockerfile \
      -t local:image-tag \
      .
    ```
+   Please note that the build only works when using the new
+   [Docker BuildKit](https://docs.docker.com/build/buildkit/) (enabled through the `DOCKER_BUILDKIT` variable).
    
-2. Set the version of gapic-generator-java
+3. Set the version of gapic-generator-java
```
Comment (JoeWang1127) [inline hermetic_build/DEVELOPMENT.md:171]: Is this the right num?

Comment (diegomarquezp) [reply hermetic_build/DEVELOPMENT.md:171]:
```suggestion
2. Set the version of gapic-generator-java
```

I didn't see the editor changed the number. Thanks for the catch

**Review by JoeWang1127 — APPROVED** [review]

File: hermetic_build/DEVELOPMENT.md [hunk 2]
```diff
@@ -160,11 +160,13 @@ python hermetic_build/library_generation/cli/entry_point.py generate \
 1. Run the following command to build the image from source
 
    ```shell
-   docker build \
+   DOCKER_BUILDKIT=1 docker build \
```
Comment (blakeli0) [inline hermetic_build/DEVELOPMENT.md:163]: Can we mention that `DOCKER_BUILDKIT=1` is only needed for local development on CloudTop but not needed in CloudBuild?

Comment (diegomarquezp) [reply hermetic_build/DEVELOPMENT.md:163]: Added a note below.

**Pull request merged**